[MISC] rename align_cfg::scoring to align_cfg::scoring_scheme

wvandertoorn opened the pull request:
Fixes https://github.com/seqan/product_backlog/issues/177.

joergi-w commented on test/performance/alignment/global_affine_alignment_benchmark.cpp:
```diff
@@ -29,7 +29,7 @@
                             seqan3::align_cfg::gap{seqan3::gap_scheme{seqan3::gap_score{-1},
                                                                       seqan3::gap_open_score{-10}}} |
-                            seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                            seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
adapt the indentation of the line below accordingly, please

joergi-w commented on test/performance/alignment/global_affine_alignment_parallel_benchmark.cpp:
```diff
@@ -34,7 +34,7 @@
                             seqan3::align_cfg::gap{seqan3::gap_scheme{seqan3::gap_score{-1},
                                                                       seqan3::gap_open_score{-10}}} |
-                            seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                            seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
same here

joergi-w commented on test/performance/alignment/global_affine_alignment_simd_benchmark.cpp:
```diff
@@ -29,7 +29,7 @@
                             seqan3::align_cfg::gap{seqan3::gap_scheme{seqan3::gap_score{-1},
                                                                       seqan3::gap_open_score{-10}}} |
-                            seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                            seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
and here

joergi-w commented on test/performance/io/format_sam_benchmark.cpp:
```diff
@@ -39,7 +39,7 @@ static std::string create_sam_file_string(size_t const n_queries)
         // align
         auto config = seqan3::align_cfg::method_global{} |
-                      seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                      seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
also here

joergi-w commented on test/unit/alignment/pairwise/fixture/global_affine_banded.hpp:
```diff
@@ -50,7 +50,7 @@ static auto dna4_01 = []()
         align_config | seqan3::align_cfg::band_fixed_size{seqan3::align_cfg::lower_diagonal{-3},
                                                           seqan3::align_cfg::upper_diagonal{8}} |
-                       seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                       seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Please adapt the indentation of the following line accordingly (+5x below).

joergi-w commented on test/unit/alignment/pairwise/fixture/local_affine_banded.hpp:
```diff
@@ -46,7 +46,7 @@ static auto dna4_01 = []()
         "AACCGGTTTAACCGGTT"_dna4,
         "ACGTCTACGTA"_dna4,
-        align_config | seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+        align_config | seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Please adapt the indentation of the following line accordingly (+5x below).

joergi-w commented on test/unit/alignment/pairwise/fixture/local_affine_unbanded.hpp:
```diff
@@ -45,7 +45,7 @@ static auto dna4_01 = []()
         "AACCGGTTTAACCGGTT"_dna4,
         "ACGTCTACGTA"_dna4,
-        align_config | seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+        align_config | seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Please adapt the indentation of the following line accordingly (+5x below).

joergi-w commented on test/unit/alignment/pairwise/fixture/semi_global_affine_banded.hpp:
```diff
@@ -43,7 +43,7 @@ static auto dna4_01_semi_first = []()
         "ACGTAAAACGT"_dna4,
         align_config_semi_seq1
-            | seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+            | seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Please adapt the indentation of the following line accordingly (+2x below).

joergi-w commented on test/unit/alignment/pairwise/fixture/semi_global_affine_unbanded.hpp:
```diff
@@ -43,7 +43,7 @@ static auto dna4_01_semi_first = []()
         "ACGTAAAACGT"_dna4,
         align_config_semi_seq1
-            | seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+            | seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Please adapt the indentation of the following line accordingly (+3x below).

joergi-w commented on test/unit/alignment/pairwise/semi_global_affine_banded_test.cpp:
```diff
@@ -35,7 +35,7 @@ struct pairwise_semiglobal_affine_banded : public ::testing::Test
                seqan3::align_cfg::gap{seqan3::gap_scheme{seqan3::gap_score{-1},
                                                          seqan3::gap_open_score{-10}}} |
-               seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+               seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Please adapt the indentation of the following line accordingly.

wvandertoorn commented:
@joergi-w thank you for pointing all the files out, I'm sorry to have overlooked the indentation shifts.
As for the naming concerns, that is a good observation. @seqan/core

rrahn commented:
We need to rename the `scoring_scheme` concept to `scoring_scheme_for` in order to reflect the concept naming rules of the standard. The rational is that scoring_scheme_for requires an additional type to check for. So I think we can keep the align_cfg naming as is.

wvandertoorn commented:
Rebased as requested to avoid some mess with the align_cfg::detail namespace (https://github.com/seqan/seqan3/pull/2020), and cleaned up the commit history.

smehringer reviewed (approved):
LGTM :)
